bloom/generators/rpm/templates/ament_python/template.spec.em

```diff
@@ -43,6 +44,18 @@ if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.
 if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi
 %py3_install -- --prefix "@(InstallationPrefix)"
 
+%if 0%{?with_tests}
+%check
+TEST_TARGET=$(ls -d * | grep -m1 "\(test\|tests\)")
+if [ -n "$TEST_TARGET" ] && %__python3 -m pytest --version; then
+# In case we're installing to a non-standard location, look for a setup.sh
+# in the install tree and source it.  It will set things like
+# CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
+if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi
+%__python3 -m pytest $TEST_TARGET || echo "RPM TESTS FAILED"
```

Comment:
Okay. I might've been hasty here as I also didn't realize that you're invoking pytest unconditionally for python tests.

Presumably this is because there's no magic tool like debhelper/dh_python for detecting and doing the right thing?

Reply:
> Presumably this is because there's no magic tool like debhelper/dh_python for detecting and doing the right thing?

I suspected there was such a mechanism in debs [earlier](https://github.com/ros-infrastructure/bloom/pull/585#discussion_r419812940), but there is no such tool in RPMs AFAIK.

My experience leads me to believe that RPM package is done much more explicitly than debs. I doubt that any "auto" behavior like we're introducing here would be allowed in official packages.